src/plugins/ccode/README.md

```diff
@@ -8,13 +8,76 @@
 - infos/status = maintained unittest nodep libc configurable
 - infos/description = Decoding/Encoding engine which escapes unwanted characters.
 
+# CCode
+
 ## Introduction
 
-When writing a C string in C code some characters cannot be expressed
-directly but have a special one letter abbreviation.
-The ccode plugin allows us to map any single escaped
-character to be replaced by another single character and vice versa.
-The user can configure this mapping.
+The `ccode` plugin replaces (escapes) any special characters with two characters:
+
+- an escape character (default: `\`) and
+- another character representing the escaped character (e.g `n` for newline)
+
+before writing a `KeySet`. The plugin undoes this modification after reading a `KeySet`.
+
+CCode provides a reasonable default configuration, using the usual escape sequences
+for C strings (e.g. `\n` for newline, `\t` for tab). You can also configure the escape character
+(`/escape`) and the mapping for special characters (`chars`).
+
+## Examples
+
+### Default Configuration
+
+```sh
+# Mount `tcl` storage plugin together with the required `base64` plugin.
+# We use the `ccode` plugin to escape special characters.
+sudo kdb mount config.tcl user/tests/ccode tcl ccode base64
+
+# Add a key value containing newline characters
+kdb set user/tests/ccode/multiline "`printf 'one\ntwo\nthree'`"
+# By default the plugin uses `\n` to escape newline characters
+grep 'multiline' `kdb file user/tests/ccode` | sed 's/[[:space:]]*//'
+#> multiline = one\ntwo\nthree
+
+# The `ccode` plugin escapes and unescapes the data. The `tcl` plugin
+# returns the unescaped values.
+kdb get user/tests/ccode/multiline
+#> one
+#> two
+#> three
+
+# Write and read a key value containing a tab character
+kdb set user/tests/ccode/tab 'Tab	Fabulous'
+kdb get user/tests/ccode/tab
+#> Tab	Fabulous
+
+# The plugin also escapes special characters inside key names
+kdb set 'user/tests/ccode/tab/t	a	b' 'Escaped Tabs'
+grep 'tab/' `kdb file user/tests/ccode` | sed 's/[[:space:]]*//'
+#> tab/t\ta\tb = Escaped Tabs
+
+# Undo modifications to database
+kdb rm -r user/tests/ccode
+sudo kdb umount user/tests/ccode
+```
+
+### Custom Configuration
+
+```sh
+# We use `%` as escape character and map the space character (hex: `20`) to the character `_`.
+sudo kdb mount config.tcl user/tests/ccode tcl base64 ccode escape=`bash -c 'printf %x \"%'` chars= chars/20=`bash -c 'printf %x \"_'`
```

Comment:
What about not requiring `chars=`?

Reply:
That sounds like a good idea, especially since I did not add `chars=` at first and had to debug why only adding `chars/20` did not work. However, for which key should we look in `ksLookupByName` then? As far as I know `ksLookupByName` does not support something like `/chars*` (glob expression) or `/chars.*` (regex) as argument.

Reply:
I think the configuration settings passed to the plugin won't be too big. So simply iterate over everything and check if it contains /chars.